### PR TITLE
Added solution to leetcode problem 211 

### LINKED ISSUE #2123: Add solution to leetcode problem 211
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

Add solution to leetcode problem 211 in c++ , python etc.

### Suggested Change

Adding solution to problem 211

### Rationale

this will help users facing difficulty in that problem.

### Urgency

High

### Acknowledgements

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository
- [X] I have read and followed the Contribution Guidelines
- [X] I have followed the code style guidelines of this project
- [X] I have checked for any existing open issues that my pull request may address
- [X] I have ensured that my changes do not break any existing functionality
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently
- [X] I have read the resources for guidance listed below
- [X] I have not used AI-generated content (e.g., ChatGPT, other AI tools)
- [X] I have not used content from existing sites (e.g., text, images)
- [X] I have followed all security rules and only shared trusted resources

### PR DESCRIPTION
**Pull request by AnshikaAg03**
## Related Issue

Fix #2123 

## Description

I have added solution of leetcode problem 211 in python and java with explaination.

## Type of PR

- [x] Documentation update
- [ ] Security enhancement

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [x] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [x] [MDX Documentation](https://mdxjs.com/docs/)
- [x] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [x] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [x] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [x] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+---
+id: Design Add and Search Words Data Structure
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md:2]: replace `id: Design Add and Search Words Data Structure` to `id: design-add-and-search-words-data-structure`

File: dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+- wordDictionary.addWord("bad");
+- wordDictionary.addWord("dad");
+- wordDictionary.addWord("mad");
+- wordDictionary.search("pad"); // return False
+- wordDictionary.search("bad"); // return True
+- wordDictionary.search(".ad"); // return True
+- wordDictionary.search("b.."); // return True
+
+### Constraints
+- 1 <= word.length <= 25
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md:40]: replace `1 <= word.length <= 25` to ``` `1 <= word.length <= 25` ```

File: dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+### Example:
+#### Input:
+
+["WordDictionary","addWord","addWord","addWord","search","search","search","search"]
+[[],["bad"],["dad"],["mad"],["pad"],["bad"],[".ad"],["b.."]]
+
+#### Output:
+[null,null,null,null,false,true,true,true]
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md:28]: replace

```md
#### Explanation:
- WordDictionary wordDictionary = new WordDictionary();
- wordDictionary.addWord("bad");
- wordDictionary.addWord("dad");
- wordDictionary.addWord("mad");
- wordDictionary.search("pad"); // return False
- wordDictionary.search("bad"); // return True
- wordDictionary.search(".ad"); // return True
- wordDictionary.search("b.."); // return True
```

to

````md
#### Explanation:
```
WordDictionary wordDictionary = new WordDictionary();
wordDictionary.addWord("bad");
wordDictionary.addWord("dad");
wordDictionary.addWord("mad");
wordDictionary.search("pad"); // return False
wordDictionary.search("bad"); // return True
wordDictionary.search(".ad"); // return True
wordDictionary.search("b.."); // return True
```
````

File: dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+- The brute force way to solve this problem is pretty simple, "having a list of words & then just for every search we would just see, if this search match any of the word in input list." [But it's not an efficient way]
+
+Let's talk about, efficient way to solve this problem. And for that we require Trie data structure a.k.a Prefix tree
+
+Let's understand Trie first,
+Trie is basically a tree that has some kind of root node & each node can have up to 26 children in this problem. Because we have lower case character from a to z. So, basically each node represent's a single character. And each node could have up to 26 additional children:
+![image](https://github.com/CodeHarborHub/codeharborhub.github.io/assets/122719744/5281ba10-236c-4bb8-bc0c-d7dbf6af300a)
+
+And basically word in this example means, let's take "a" having child "b" -> with "c", so that's will be a single word. And if we insert the word "abc" in our trie, so basically how's it looks like:
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/0211-Design Add and Search Words Data Structure.md:62]: replace `->` to ``` `->` ```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by AnshikaAg03** [comment]
image
how to create a single pull request for both of these changes @Ajay-Dhangar

**Comment by Ajay-Dhangar** [comment]
> image how to create a single pull request for both of these changes @Ajay-Dhangar

You can be create one PR and assign 2 issues like as reference I will merge and added for both

**Pull request merged**